src/environment.cpp

```diff
@@ -271,6 +271,210 @@ ABMWithState::ABMWithState(ActiveBlockModifier *abm_):
 }
 
 /*
+	LBMManager
+*/
+
+LBMContentMapping::~LBMContentMapping()
+{
+	for (std::vector<LoadingBlockModifierDef *>::iterator it = lbm_list.begin();
+			it != lbm_list.end(); ++it) {
+		delete *it;
+	}
+}
+
+void LBMContentMapping::addLBM(LoadingBlockModifierDef *lbm_def, IGameDef *gamedef)
+{
+	// Add the lbm_def to the LBMContentMapping.
+	// Unknown names get added to the global NameIdMapping.
+	INodeDefManager *nodedef = gamedef->ndef();
+
+	lbm_list.push_back(lbm_def);
+
+	for (std::set<std::string>::const_iterator it = lbm_def->trigger_contents.begin();
```

Comment:
Unicity will be tested at insertion, and then... at server start. std::set is not pertinent here, std::vector will be faster